Blockzilla/IntroViewController.swift

```diff
@@ -26,6 +26,104 @@ struct IntroViewControllerUX {
     static let FadeDuration = 0.25
 }
 
+protocol PageControlDelegate {
+    func selectedIndex(_ index:Int)
+}
+
+class PageControl: NSObject {
+    
+    var currentPage = 0
+    var numberOfPages = 0
+    var hidesForSinglePage = false
+    
+    var stack : UIStackView? = nil
+    var delegate: PageControlDelegate?
+    
+    init(numberOfPages: Int) {
+        super.init()
+
+        var buttonArray: [UIButton]!
```

Comment:
Prefer empty arrays  to optional or implicit unwrap,`var buttonArray = [UIButton]()`. LMK if you want more info on this.